### PR TITLE
Replace the TestMailServer to GreenMail server

### PR DESCRIPTION
**Pull request by rachana-dani**
### Description
[Describe what this change achieves]
Switched TestMailServer to GreenMail server

### Issues Resolved
[List any issues this PR will resolve]
This is to stop using deprecated test smtp server

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Hailong-am** [comment]
@rachana-dani can you fix the DCO check? you need add `-s` flag with git commit command

File: notifications/core/src/test/kotlin/org/opensearch/notifications/core/smtp/SmtpEmailTests.kt [hunk 1]
```diff
@@ -14,28 +17,29 @@ import org.opensearch.notifications.core.transport.SmtpDestinationTransport
...
 class SmtpEmailTests {
 
-    internal companion object {
-        private const val smtpPort = 10255 // use non-standard port > 1024 to avoid permission issue
-        private val smtpServer = TestMailServer.smtp(smtpPort)
+    private lateinit var greenMail: GreenMail
+
+    @BeforeEach
+    fun setUpServer() {
+        greenMail = GreenMail(ServerSetupTest.SMTP)
```
Comment (Hailong-am) [inline notifications/core/src/test/kotlin/org/opensearch/notifications/core/smtp/SmtpEmailTests.kt:28]: what's the default port does test server used?

**Review by zhichao-aws — COMMENTED** [review]
Is there a method to verify whether the GreenMail server recieve the notification?

File: notifications/core/src/test/kotlin/org/opensearch/notifications/core/smtp/SmtpEmailTests.kt (hunk 1 above)
Comment (rachana-dani) [reply notifications/core/src/test/kotlin/org/opensearch/notifications/core/smtp/SmtpEmailTests.kt:28]: This is typically 3025

**Comment by rachana-dani** [comment]
> Is there a method to verify whether the GreenMail server recieve the notification?

There is a method greenMail.receivedMessages. I am checking size of the returned object

**Comment by rachana-dani** [comment]
> > Is there a method to verify whether the GreenMail server recieve the notification?
> 
> There is a method greenMail.receivedMessages. I am checking size of the returned object

I have addressed all the comments. Please take a look.

**Comment by rachana-dani** [comment]
> > > Is there a method to verify whether the GreenMail server recieve the notification?
> > 
> > 
> > There is a method greenMail.receivedMessages. I am checking size of the returned object
> 
> I have addressed all the comments. Please take a look.

@Hailong-am @zhichao-aws all the checks have passed.

File: notifications/core/build.gradle [hunk 2]
```diff
@@ -161,6 +161,10 @@ dependencies {
             'io.mockk:mockk-agent-jvm:1.11.0'
     )
     testImplementation 'org.springframework.integration:spring-integration-mail:5.5.0'
+    // https://mvnrepository.com/artifact/com.icegreen/greenmail
+    testImplementation group: 'com.icegreen', name: 'greenmail', version: '1.5.0'
```
Comment (Hailong-am) [inline notifications/core/build.gradle:165]: is it possible we use latest version of greenmail? I do see there have two CVEs with version 1.5.0 https://mvnrepository.com/artifact/com.icegreen/greenmail/1.5.0

**Comment by rachana-dani** [comment]
Can you please add a HackoberFest2023 label and help me merge this PR?

**Review by Hailong-am — APPROVED** [review]
Looks good to me

File: notifications/core/build.gradle [hunk 3]
```diff
@@ -161,6 +161,8 @@ dependencies {
             'io.mockk:mockk-agent-jvm:1.11.0'
     )
     testImplementation 'org.springframework.integration:spring-integration-mail:5.5.0'
+    // https://mvnrepository.com/artifact/com.icegreen/greenmail
+    testImplementation group: 'com.icegreen', name: 'greenmail', version: '1.6.4'
```
Comment (Hailong-am) [inline notifications/core/build.gradle:165]:
```suggestion
    testImplementation group: 'com.icegreen', name: 'greenmail', version: '1.6.14'
```

Comment (rachana-dani) [reply notifications/core/build.gradle]: I have updated this version. Pls R&A

**Review by Hailong-am — APPROVED** [review]

**Review by zhichao-aws — APPROVED** [review]

**Comment by SuZhou-Joe** [comment]
#789

**Pull request merged**